src/input/keyboard/mod.rs

```diff
@@ -707,6 +733,21 @@ impl<'a, D: SeatHandler + 'static> KeyboardInnerHandle<'a, D> {
         };
     }
 
+    /// Iterate over the currently pressed keys.
+    pub fn with_pressed_keysyms<F, R>(&self, f: F) -> R
+    where
+        F: FnOnce(Vec<KeysymHandle<'_>>) -> R,
+        R: 'static,
+    {
+        let handles = self
+            .inner
+            .pressed_keys
+            .iter()
+            .map(|code| self.keysym_handle(*code))
+            .collect();
+        f(handles)
+    }
```

Comment:
Just to be sure, this is callback based for the sake of consistency with KeyboardHandle (non inner)?

Reply:
yes